src/components/Complain.js

```diff
@@ -34,7 +34,7 @@ const Complain = () => {
   const onSubmit = (e) => {
     e.preventDefault();
     const isvalid = validationFunc(inputErr, state, setInputErr);
-    if (isvalid.length === 0) {
+    if (!isvalid) {
```

Comment:
If it's not valid, then why you're calling DB API?
DB API should be called only if form is valid